controllers/staff.js

```diff
@@ -50,6 +50,37 @@ LIMIT $4;
 
 const psGetMinMaxDate = new PS({ name: 'get-min-max-date', text: 'SELECT min(deliveredTime) as minDate, max(deliveredTime) as maxDate FROM Orders;' });
 
+const psGetPromoStats = new PS({ name: 'get-promo-stats', text: `
+WITH PromoDuration AS (
+SELECT pid, startDate, endDate, (endDate - startDate + 1) as duration
+FROM RestaurantPromos natural join Promotions
+WHERE rid = $1)
+
+SELECT pid, duration, count(*) as numOrders
+FROM PromoDuration natural join Orders
+WHERE orderTime >= startDate AND orderTime <= endDate
+GROUP BY pid, duration; 
+`});
+
+const psGetAllOrders = new PS({ name: 'get-all-orders', text: 
+`
+WITH OrderRestaurantPair AS (
+    SELECT DISTINCT rid, oid
+    FROM Collates natural join Orders
+),
+RidToName AS (
+    SELECT uid as riderId, name as rname
+    FROM Riders natural join Users
+),
+CidToName AS (
+    SELECT uid as customerId, name as cname
+    FROM Customers natural join Users
+)
+SELECT oid, rname, cname, orderTime, deliveredTime, finalPrice, (SELECT string_agg(fname, ', ') FROM COLLATES C WHERE C.oid = X.oid) as itemsOrdered
```

Comment:
correct me if I'm wrong, the food is returned as a comma seperated string right? I dont this is good. What if you have a food entry named: "Egg, Chicken and rice", the front end doesnt know whether theres 2 types of food or 1 type of food

Reply:
ok ill separate it with `|` instead

Reply:
And if the name cotains | ?

Reply:
Please use array_agg(fname) instead